gramex/data.py

```diff
@@ -1481,23 +1533,39 @@ def _filter_mongodb(url, controls, args, database=None, collection=None, query=N
         for col, val in data.iloc[0].iteritems():
             if type(val) in {bson.objectid.ObjectId}:
                 data[col] = data[col].map(str)
+
     return data
 
 
 def _delete_mongodb(url, controls, args, meta=None, database=None, collection=None, query=None,
                     **kwargs):
     table = _mongodb_collection(url, database, collection, **kwargs)
-    meta_cols = pd.DataFrame(list(table.find().limit(100)))
-    query = _mongodb_query(args, meta_cols)
+    query = _mongodb_query(args, table)
     result = table.delete_many(query)
     return result.deleted_count
 
 
 def _update_mongodb(url, controls, args, meta=None, database=None, collection=None, query=None,
                     id=[], **kwargs):
+    from dateutil.parser import parse
+
     table = _mongodb_collection(url, database, collection, **kwargs)
-    query = _mongodb_query(args, id)
-    values = {key: val[0] for key, val in args.items()}
+    query = _mongodb_query(args, table, id=id)
+    results = table.find(query).limit(1)
+
+    if not results.count():
+        return 0
+
+    values = {key: val[0] for key, val in dict(args).items() if key not in id}
+
+    for key, val in values.items():
+        convert = _convert_datatype(results, key)
+
+        if convert == bool:
+            values[key] = 0 if val in ['0', 'false', 'False'] else 1
+
```

Comment:
Reduce the number of empty newlines. Group into logical chunks, but slighly larger chunks than you have now :-)